File: WIT.Portal/src/app/entities/product-feature.entity.ts
import { BaseEntity } from './base.entity';
import { ListBase, GridInfo } from './grid-info.entity';

export class ProductFeature extends BaseEntity {
    public productFeatureID: number;
    public code: string;
    public name: string;
    public description: string;

    constructor()
    constructor(rhs: ProductFeature)
    constructor(rhs?: ProductFeature) {
        super(rhs);
        if (!rhs) {
            this.productFeatureID = 0;
        }
    }

    public get caption() {
        return this.code + '-' + this.name || '';
    }

    public get isValid() {
        return ((this.code || "") !== "");
    }
}

export class ProductFeatureList extends ListBase {
    public items: Array<ProductFeature>;
    public code: string;
    public name: string;

    constructor()
    constructor(rhs: ProductFeatureList)
    constructor(rhs?: ProductFeatureList) {

        super(rhs);

        if (!this.items || !Array.isArray(this.items)) {
            this.items = new Array<ProductFeature>();
        } else {
            this.items = this.items.map(v => new ProductFeature(v));
        }
    }
}

File: WIT.Business/Transactions/ProductFeatureEntity.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Collections;
using WIT.Data.Models;

namespace WIT.Business.Entities
{
    public class ProductFeatureItem : BaseEntity
    {
        public int ProductFeatureID { get; set; }
        public string Code { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }

        public ProductFeatureItem() { }
        public ProductFeatureItem(ProductFeature rhs)
        {
            CreatedOn = rhs.CreatedOn;
            UpdatedOn = rhs.UpdatedOn;
            CreatedBy = rhs.CreatedBy;
            UpdatedBy = rhs.UpdatedBy;

            ProductFeatureID = rhs.ProductFeatureID;
            Code = rhs.Code;
            Name = rhs.Name;
            Description = rhs.Description;
        }
    }

    public class ProductFeatureList
    {
        public GridInfo GridInfo { get; set; }
        public List<ProductFeatureItem> Items { get; set; }

        public string Code { get; set; }
        public string Name { get; set; }

        public ProductFeatureList()
        {
            Items = new List<ProductFeatureItem>();
            GridInfo = new GridInfo();
        }
    }
}
